src/Symfony/Component/Notifier/Bridge/Mastodon/MastodonTransport.php

```diff
@@ -0,0 +1,139 @@
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Quentin Dequippe <quentin@dequippe.tech>
+ *
+ * @see https://docs.joinmastodon.org
+ */
+final class MastodonTransport extends AbstractTransport
+{
+    public function __construct(#[\SensitiveParameter] private readonly string $accessToken, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mastodon://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof MastodonOptions);
+    }
+
+    public function request(string $method, string $url, array $options): ResponseInterface
+    {
+        $url = sprintf('https://%s%s', $this->getEndpoint(), $url);
+
+        $options['auth_bearer'] = $this->accessToken;
+
+        return $this->client->request($method, $url, $options);
+    }
+
+    /**
+     * @see https://docs.joinmastodon.org/methods/statuses/
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
+        }
+
+        $options = $message->getOptions()?->toArray() ?? [];
+        $options['status'] = $message->getSubject();
+        $response = null;
+
+        try {
+            if (isset($options['attach'])) {
+                $options['media_ids'] = $this->uploadMedia($options['attach']);
+                unset($options['attach']);
+            }
+
+            $response = $this->request('POST', '/api/v1/statuses', ['json' => array_filter($options)]);
+            $statusCode = $response->getStatusCode();
+            $result = $response->toArray(false);
+        } catch (ExceptionInterface $e) {
+            if (null !== $response) {
+                throw new TransportException($e->getMessage(), $response, 0, $e);
+            }
+
+            throw new RuntimeException($e->getMessage(), 0, $e);
+        }
+
+        if (200 !== $statusCode) {
+            throw new TransportException(sprintf('Unable to post the Mastodon message: "%s" (%s).', $result['error_description'], $result['error']), $response);
+        }
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId($result['id']);
+
+        return $sentMessage;
+    }
+
+    private function uploadMedia(array $media): array
+    {
+        $responses = [];
+
+        /**
+         * @var File        $file
+         * @var File|null   $thumbnail
+         * @var string|null $description
+         * @var string|null $focus
```

Comment:
better put this as a shape on the argument:
`@param array<array{file: File, thumbnail: File|null, description: string|null, focus: string}> $media`